sql/core/src/test/resources/sql-tests/inputs/cast-format.sql

```diff
@@ -0,0 +1,53 @@
+CREATE TEMPORARY VIEW strings(s1, s2, s3, s4, s5, s6, s7) AS VALUES (
+  null,
+  '2019',
+  '2019 05',
+  '2019 05 28',
+  '2019 05 28 17 03 53',
+  '2019 05 28 17 03 53 123',
+  '2019 05 28 17 03 53 123456'
+);
+
+SELECT
+    CAST(s1 AS DATE FORMAT 'yyyy'),
+    CAST(s2 AS DATE FORMAT 'yyyy'),
+    CAST(s3 AS DATE FORMAT 'yyyy MM'),
+    CAST(s4 AS DATE FORMAT 'yyyy MM dd'),
+    CAST(s2 AS DATE FORMAT 'invalid')
+FROM strings;
+
+SELECT
+    CAST(s1 AS TIMESTAMP FORMAT 'yyyy'),
+    CAST(s2 AS TIMESTAMP FORMAT 'yyyy'),
+    CAST(s3 AS TIMESTAMP FORMAT 'yyyy MM'),
+    CAST(s4 AS TIMESTAMP FORMAT 'yyyy MM dd'),
+    CAST(s5 AS TIMESTAMP FORMAT 'yyyy MM dd HH mm ss'),
+    CAST(s6 AS TIMESTAMP FORMAT 'yyyy MM dd HH mm ss SSS'),
+    CAST(s7 AS TIMESTAMP FORMAT 'yyyy MM dd HH mm ss SSSSSS'),
```

Comment:
I don't think we need to test `TimestampFormatter` on different patterns here. Instead it is better to test corner cases like:
```
CAST(... AS TIMESTAMP FORMAT '')
CAST(... AS TIMESTAMP FORMAT)
...
```

Reply:
Thanks, I've changed it a bit, now covers the above cases.